tests/next_tests/unit_tests/iterator_tests/ir_utils_test.py/test_misc.py

```diff
@@ -0,0 +1,62 @@
+# GT4Py - GridTools Framework
+#
+# Copyright (c) 2014-2024, ETH Zurich
+# All rights reserved.
+#
+# Please, refer to the LICENSE file in the root directory.
+# SPDX-License-Identifier: BSD-3-Clause
+
+import pytest
+
+from gt4py.next.iterator.ir_utils import ir_makers as im, misc
+from gt4py.next.iterator.transforms import inline_lambdas
+
+
+@pytest.mark.parametrize(
+    "expr,  expected_expr",
+    [
+        (
+            im.let("a", "b")(im.make_tuple(im.tuple_get(0, "a"), im.tuple_get(42, "a"))),
+            im.ref("b"),
+        ),
+        (
+            im.tuple_get(1, im.ref("a")),
+            im.ref("a"),
+        ),
+        (
+            im.make_tuple(im.tuple_get(0, im.ref("a")), im.tuple_get(42, im.ref("a"))),
+            im.ref("a"),
+        ),
+        (
+            im.make_tuple(
+                im.tuple_get(1, im.ref("a")),
+                im.make_tuple(im.tuple_get(2, im.ref("a")), im.tuple_get(0, im.ref("a"))),
+            ),
+            im.ref("a"),
+        ),
+        (
+            im.tuple_get(3, im.tuple_get(0, im.ref("a"))),
+            im.ref("a"),
+        ),
+        (
+            im.plus(im.ref("a"), im.ref("b")),
+            im.plus(im.ref("a"), im.ref("b")),
+        ),
+        (
+            im.call("as_fieldop")(im.ref("a")),
+            im.call("as_fieldop")(im.ref("a")),
```

Comment:
How about the scan example here: `as_fieldop(scan(λ(state, val) → {val, state[0]+val}))(inp)[1]`?